packages/ui/src/components/Button/Button.tsx

```diff
@@ -29,17 +36,27 @@ const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
       throw new Error("Icon is required when mode is set to 'cta'");
     }
 
-    const styles = useCommonButtonLinkStyles({
-      mode,
-      color,
-      block,
-      disabled,
-    });
-
     return (
       // eslint-disable-next-line react/button-has-type
-      <button ref={ref} disabled={disabled} {...styles} {...rest}>
-        <ButtonLinkContent mode={mode} icon={icon} utilityIcon={utilityIcon}>
+      <button
+        ref={ref}
+        disabled={disabled}
+        {...getBaseStyle({ mode })}
+        {...getBackgroundColor({ mode, color, disabled })}
+        {...getBorderColor({ mode, color, disabled })}
+        {...getDisplay({ mode, block })}
+        {...getFontColor({ mode, color, disabled })}
+        {...getFontWeight({ mode })}
+        {...getShadow({ mode, disabled })}
```

Comment:
I'm not against moving these functions into their own files, but I think it was nice to have a single function defined somewhere which combines them.

Reply:
Separating them was mostly for my ADHD brain - I couldn't focus on the existing file you guys were using because there was so much happening. Happy to add a consolidation function if you think it optimizes things.

Reply:
Done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/7946/commits/b65ccf6fa9b5e37028d9f7d74d79d8c7763fbc56